### ISSUE TITLE
mono fails to build on aarch64-darwin

### ISSUE THREAD

**Issue by babbaj** [NixOS/nixpkgs#173216]
### Steps To Reproduce
1. build mono on aarch64-darwin

### Build log
https://gist.github.com/babbaj/d1e86b0dd4bbb16664cfdc4f094518f1

### Additional context
This is blocking any package made with `buildDotnetModule` from building with dotnet 6.
relevant code: https://github.com/mono/mono/blob/5d2e3bc3b3c8184d35b2f7801e88d96470d367c4/mcs/build/profiles/build.make#L92

### Notify maintainers

@thoughtpolice @obadz @vrthra 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
@soybook:~ ❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 21.4.0, macOS 12.3.1`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.0`
 - channels(babbaj): `""`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/0y06zg1h1ks4ld6wizg3acraasikgqkk-57y6djpqdmjglkdzgrbsdyjzqkn48ff1-source`
```

**Comment by jiegec** [comment]
See issue #151720

**Comment by jiegec** [comment]
Fixed by #181051
